### DISCUSSION TITLE
After migration to non-archival relay, not ingesting records from some independent PDSes

### DISCUSSION THREAD

**Discussion by snarfed** [bluesky-social/atproto#3094] [Dev Debugging] [answered]
Hi all! Congrats on keeping the ship afloat during last week's relay migration and firefighting (https://github.com/bluesky-social/atproto/discussions/3036, [more](https://bsky.app/profile/bnewbold.net/post/3lbirrqme2s2w)). Hope you all have recovered and gotten some sleep!

Since then, some independent PDSes' records are getting ingested and visible in the main appview and bsky.app, but not all. Specifically Bridgy Fed's (atproto.brid.gy) records stopped getting ingested roughly four days ago, 11/20 21:28 UTC, atproto.brid.gy subscribeRepos seq 4317000 ish. Tracked on our side in https://github.com/snarfed/bridgy-fed/issues/1520#issuecomment-2492534380. I don't know if this is happening to other independent PDSes too, or just us.

Example missing record: `at://did:plc:bg5udl25mvzg3rt7l5n2hzet/app.bsky.feed.post/3lbhz4myfsjg2`, atproto.brid.gy seq 4368917, emitted sometime around 11/21 17:41:08 UTC, not visible on https://bsky.app/profile/did:plc:bg5udl25mvzg3rt7l5n2hzet

I've talked to a few people on the team about this on and off; filing here just to have a shared place to track. Lmk if I can help. #hugops!

**Comment by bnewbold** [comment]
A number of things went wrong on 2024-11-21, which impacted both Bluesky-hosted PDS instances and external PDS instances alike.

I'm not aware of any *current* problems with *new* events and records being emitted in the past 24 hours, from any PDS instances in the network. Are you seeing any problems of that type, or only broken/missing content from last week?

If you have a set of accounts (DIDs) missing prior content, we have a mechanism to re-index things at the AppView.

A hack to get things to re-index more broadly is to emit updates or create/delete cycles on individual records. This might not make sense for posts, but has helped a few folks get profile updates unblocked (eg, updates to add and then remove whitespace from profile descriptions).

**Reply by glyph** [reply]
I am just a user of the bridge, so I apologize if this is too imprecise of a report, but I would expect e.g.: https://mastodon.social/@glyph/113541915554138391 (A post I wrote *after* reading your comment here) to be visible at https://bsky.app/profile/glyph.mastodon.social.ap.brid.gy

So from my perspective, whatever incident provoked this is ongoing.

**Reply by snarfed** [reply]
> I'm not aware of any _current_ problems with _new_ events and records being emitted in the past 24 hours, from any PDS instances in the network. Are you seeing any problems of that type, or only broken/missing content from last week?

@bnewbold oh interesting! Thanks for the nudge, that explanation makes sense given the switch to non-archival, but atproto.brid.gy does still seem stuck somewhat PDS-wide. I tried a few different types of events and records just now, and I'm not seeing any of them:

* [new post](https://indieweb.social/@snarfed/113541861287492795) for https://bsky.app/profile/fedi.test.snarfed.org : `did:plc:bg5udl25mvzg3rt7l5n2hzet`, seq 4571157, `#commit`, `at://did:plc:bg5udl25mvzg3rt7l5n2hzet/app.bsky.feed.post/3lbqqwqyxr232`
* deactivated https://bsky.app/profile/vg.no.web.brid.gy : `did:plc:dvca4532rjpurs2keeiysr6m`, seq 4572947, `#account status: deactivated`
* new account, `did:plc:ch3dagebvx5vjhooi26ap3gg`: seq 4573023 `#identity`, seq 4573024 `#account`

**Reply by snarfed** [reply]
I am seeing _some_ new posts now! Eg https://bsky.app/profile/vg.no , https://bsky.app/profile/bkkraindarbot.misskey.winscloud.net.ap.brid.gy

Not all, though, and not account deactivations or new accounts.

**Reply by bnewbold** [reply]
Ok, this is news to us, possibly a miscommunication on our part, we thought this was all resolved last Friday afternoon.

Will prioritize investigating these examples.

**Reply by devinivy** [reply, marked as answer]
I think we're properly kicked into gear now, and should be fully up to date within a few hours if all goes to plan.  The ingestion rate was essentially too slow and ended-up continuously moving forward but falling behind.  Will be monitoring it closely today 👍

**Reply by snarfed** [reply]
Thank you @devinivy! I'm seeing new records take effect now too. Great news!

Do you all know the rough time period of past records that still aren't ingested? I'd like to be able to communicate to my users, and especially to know which opt-out requests I need to redo since the account deactivations didn't take effect.

**Reply by devinivy** [reply]
Here's the relevant timeline I was able to piece together on this (times are in EST):
 - Weds 20th 04:30PM: bluesky core services cut over to non-archival relay ("na-relay").
 - Weds 20th 04:45PM: first bridgy event is seen by na-relay. na-relay starts catching-up at >5x for several hours but soon hits rate limits and ingestion slows down. bridgy experiences some intermittent disconnections throughout the evening.
 - Thurs 21st 07:45PM: the firehose (bsky.network) cuts over to na-relay in response to ongoing issues.
 - Thurs 21st 09:20PM: the firehose starts properly passing through account events (e.g. deactivations) to firehose consumers.
   - bluesky core services have been reliably observing account events.
 - Thurs 21st 11:00PM: bridgy is now reliably connected, but doesn't catch-up due to rate limiting.
 - Mon 25th 11:30AM: rate limits are lifted on bridgy, and it rapidly catches-up.
 - Mon 25th 02:30PM: bridgy is fully caught-up and operation is normal.


Based on this, the impact that I would anticipate is that bridgy events are behind "live" to various degrees from Weds 20th 04:30PM until Mon 25th 02:30PM.  Between Thurs 21st 07:45PM and 09:20PM firehose consumers such as bridgy would not have observed account events (e.g. deactivations) from bsky.network, though Bluesky core services did observe them.

This does leave a hole in my understanding since you are able to cite at least [this post](https://atproto-browser.vercel.app/at/did:plc:bg5udl25mvzg3rt7l5n2hzet/app.bsky.feed.post/3lbhz4myfsjg2) which is not indexed by the Bluesky app.  I thought we were able to maintain continuity from bridgy, and in total we've processed the number of events I would expect from bridgy since the cutover to na-relay, despite the intermittent rate limiting or disconnection.  Is it possible we fell out of bridgy's backfill window, causing a gap?  I will speak to the team about it and see if I could have missed something.  _Very much_ appreciate your patience here.

**Reply by snarfed** [reply]
Thank you for the details @devinivy! That all makes sense, both your timeline and the gap in understanding so far. Definitely possible that you all fell out of BF's rollback window at some point, it was only 50k seqs for a while, which was a bit less than a day. I bumped it up to 200 seqs at roughly 11/22 3p PT; I can extend it further if it'll help you all backfill.

**Comment by lucajet** [comment]
is there anything we can do to help ?

this is a follow attempt from yesterday

https://fed.brid.gy/log?path=%2Fqueue%2Fwebmention%2C%2Finbox&start_time=1732470514&key=https%3A%2F%2Fmastodon.uno%2Fbdfb489a-cfb8-47ae-879f-78eaac877099

**Comment by gunchleoc** [comment]
Some more test data from an account that's not very busy, in case it will come in useful:

I signed up with https://ailbhean.co-shaoghal.net/@gunchleoc 3 days ago and I got the notification from Bridgy immediately, but it took a long while for the profile to be created at https://bsky.app/profile/gunchleoc.ailbhean.co-shaoghal.net.ap.brid.gy - might have been a few hours if I remember correctly.

Today, https://bsky.app/profile/angeidheal.bsky.social followed me back as @/@angeidheal.bsky.social@bsky.brid.gy and it went through without a hitch. He hasn't posted since then, so no test data there.

Most of my posts are still missing on BlueSky though.

**Reply by gunchleoc** [reply]
My posts have arrived now

**Comment by juanro49** [comment]
I think that new relay have a 2 days of lag of indexing PDS
